File: android-app/src/main/java/com/jaa/likeastarappmpp/units/SettingsTableDataFactory.kt
package com.jaa.likeastarappmpp.units

import com.jaa.library.feature.settings.model.SettingsRowData
import com.jaa.library.feature.settings.presentation.SettingsTableDataFactoryInterface
import com.jaa.likeastarappmpp.SettingsRow
import dev.icerock.moko.units.TableUnitItem

class SettingsTableDataFactory : SettingsTableDataFactoryInterface {

    override fun createSettingsRow(
        itemId: Long,
        setting: SettingsRowData,
        listener: SettingsTableDataFactoryInterface.OnchangeActiveStateListener
    ): TableUnitItem {
        return SettingsRow().apply {
            this.itemId = itemId
            this.setting = setting
            this.listener = listener
        }
    }

}

File: ios-app/src/view/SettingsTableDataFactory.swift

import UIKit
import MultiPlatformLibraryUnits
import MultiPlatformLibrary
import MultiPlatformLibraryMvvm

class SettingsTableDataFactory: SettingsTableDataFactoryInterface {
    
    func createSettingsRow(itemId: Int64, setting: SettingsRowData, listener: SettingsTableDataFactoryInterfaceOnchangeActiveStateListener) -> TableUnitItem {
        return UITableViewCellUnit<SettingsTableViewCell>(
            data: SettingsTableViewCell.CellModel(
                settingData: setting,
                listener: listener
            ),
            itemId: itemId,
            configurator: nil
        )
    }

}
